**Feature:** Add spinner feature to card tabs (#877)
contiamo/operational-ui: src/Card/README.md

  tabs={[
    {
      name: "Tab 1",
      children: (
        <Button
          onClick={() => {
            setState(() => ({
              isTab1Loading: true,
            }))
            setTimeout(() => {
              setState(() => ({
                isTab1Loading: false,
              }))
            }, 1500)
          }}
        >
          Refresh
        </Button>
      ),

>       loading: state.isTab1Loading ? true : undefined,

Reviewer: Can this not just be

```suggestion
      loading: state.isTab1Loading,
```

?

>       icon: state.isTab1Loading ? undefined : "Yes",

Reviewer: I think this is strange because I'd expect the component to just replace any icon with a spinner _automatically_ instead of having to explicitly set this.

Author: What are you suggesting exactly? No discriminated interfaces? Some in-between?

Author: We're removing discriminated interfaces as per offline discussion :)

  tabs={[
    {
      name: "Tab 1",
      children: (
        <Button
          onClick={() => {
            setState(() => ({
              isTab1Loading: true,
            }))
            setTimeout(() => {
              setState(() => ({
                isTab1Loading: false,
              }))
            }, 1500)
          }}
        >
          Refresh
        </Button>
      ),
      loading: state.isTab1Loading,

>       // Icon name is not taken into consideration when loading

Reviewer: ```suggestion
      // The icon is replaced by a spinner when loading.
```

  tabs={[
    {
      name: "Tab 1",
      children: (
        <Button
          onClick={() => {
            setState(() => ({
              isTab1Loading: true,
            }))
            setTimeout(() => {
              setState(() => ({
                isTab1Loading: false,
              }))
            }, 1500)
          }}
        >
          Refresh
        </Button>
      ),
      icon: "User",

>       ...(state.isTab1Loading

Reviewer: This is a little hard to read. Can we express this in a simpler way for newcomers reading our docs?
